.github/workflows/generate-and-format.yml

```diff
@@ -0,0 +1,46 @@
+name: Generate and Format
+
+on:
+  pull_request:
+    paths:
+      - "**.go"
```

Comment:
You should also consider yaml files. For instance metrics.yaml change should trigger code-generation but I don't think it does.